addons/portal_rating/models/mail_message.py

```diff
@@ -40,12 +40,26 @@ def _portal_message_format(self, properties_names, options=None):
             for rating in related_rating
         }
 
+        rating_stats_by_model = {}
+        for message in self:
+            if not message.model:
+                continue
+            if message.model not in rating_stats_by_model:
+                rating_stats_by_model[message.model] = {}
+            if message.res_id and hasattr(self.env[message.model], 'rating_get_stats'):
+                rating_stats_by_model[message.model][message.res_id] = (
+                    self.env[message.model].browse(message.res_id).sudo().rating_get_stats()
+                )
+            else:
+                rating_stats_by_model[message.model][message.res_id] = None
+
         for message, values in zip(self, vals_list):
             values["rating"] = message_to_rating.get(message.id, {})
-
-            record = self.env[message.model].browse(message.res_id)
-            if hasattr(record, 'rating_get_stats'):
-                values['rating_stats'] = record.sudo().rating_get_stats()
+            values["rating_stats"] = (
+                rating_stats_by_model.get(message.model, {}).get(message.res_id)
+                if message.model
+                else {}
+            )
```

Comment:
Do we want this now? Would say it makes more sense not to set the key at all if there's no entry in the dict
```suggestion
            if rating_stats := rating_stats_by_model.get(message.model, {}).get(message.res_id):
                 values["rating_stats"] = rating_stats
```